src/window/city.c

```diff
@@ -500,15 +503,43 @@ static void handle_hotkeys(const hotkeys *h)
     }
     if (h->mothball_toggle) {
         int building_id = map_building_at(widget_city_current_grid_offset());
-        if (building_id) {
+        building *b = building_main(building_get(building_id));
+        if (building_id && model_get_building(b->type)->laborers) {
+                building_mothball_toggle(b);
+                if (b->state == BUILDING_STATE_IN_USE) {
+                    city_warning_clear_all();
+                    city_warning_show(WARNING_DATA_MOTHBALL_OFF, NEW_WARNING_SLOT);
+                } else if (b->state == BUILDING_STATE_MOTHBALLED) {
+                    city_warning_clear_all();
+                    city_warning_show(WARNING_DATA_MOTHBALL_ON, NEW_WARNING_SLOT);
+                }
+        }
+    }
+    if (h->storage_order) {
+        int grid_offset = widget_city_current_grid_offset();
+        int building_id = map_building_at(grid_offset);       
+        if (building_id) {   
             building *b = building_main(building_get(building_id));
-            building_mothball_toggle(b);
-            if (b->state == BUILDING_STATE_IN_USE) {
-                city_warning_clear_all();
-                city_warning_show(WARNING_DATA_MOTHBALL_OFF, NEW_WARNING_SLOT);
-            } else if (b->state == BUILDING_STATE_MOTHBALLED) {
-                city_warning_clear_all();
-                city_warning_show(WARNING_DATA_MOTHBALL_ON, NEW_WARNING_SLOT);
+            if (b->type == BUILDING_WAREHOUSE || 
```

Comment:
Instead of this gigantic `if`, can you create a static function like this please:

```
static int has_storage_orders(building_type type) {
    return type == BUILDING_WAREHOUSE_SPACE ||
    /*
     * ...
     * Etc for all buildings in the if
     */
}
```

Then you'll only need for this line:

```
if (has_storage_orders(b->type)) {
```

Much cleaner!